index.js

```diff
@@ -1,37 +1,72 @@
 import { createSelector } from 'reselect';
 
+// @TODO Move it into a separate file and test it
+function flatCacheObjectCreator() {
```

Comment:
Done! I was trying to keep the core as light as possible, and introducing classes with additional cache creators only (maybe in a separate file). But let's keep this optimization for the end. Feel free to push directly on this branch or go with PR, as you prefer. Cheers!